internal/source/https.go

```diff
@@ -0,0 +1,95 @@
+package source
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	"github.com/go-git/go-billy/v5/memfs"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+// http is a bundle source that sources bundles from the specified url.
+type HTTPS struct {
+	client.Reader
+	SecretNamespace string
+}
+
+// Unpack unpacks a bundle by requesting the bundle contents from a specified URL
+func (b *HTTPS) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
+	if bundle.Spec.Source.Type != rukpakv1alpha1.SourceTypeHTTPS {
+		return nil, fmt.Errorf("cannot unpack source type %q with %q unpacker", bundle.Spec.Source.Type, rukpakv1alpha1.SourceTypeHTTPS)
+	}
+
+	url := bundle.Spec.Source.HTTPS.URL
+	action := fmt.Sprintf("%s %s", http.MethodGet, url)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("create http request %q for bundle content: %v", action, err)
+	}
+	var userName, password string
+	if bundle.Spec.Source.HTTPS.Auth.Secret.Name != "" {
+		userName, password, err = b.getCredentials(ctx, bundle)
+		if err != nil {
+			return nil, err
+		}
+		req.SetBasicAuth(userName, password)
+	}
+
+	httpClient := http.Client{Timeout: 10 * time.Second}
+	if bundle.Spec.Source.HTTPS.Auth.InsecureSkipVerify {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		httpClient.Transport = tr
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("%s: http request for bundle content failed: %v", action, err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%s: unexpected status %q", action, resp.Status)
+	}
+
+	var memFS = memfs.New()
+	file, err := memFS.Create(filepath.Join("manifests", "chart"))
+	if err != nil {
+		return nil, fmt.Errorf("creating filesystem: %s", err)
+	}
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("writing chart file: %s", err)
+	}
+	err = file.Close()
+	if err != nil {
+		return nil, fmt.Errorf("closing chart file: %s", err)
+	}
```

Comment:
The https source shouldn't be helm-specific. My thought would be to assume that all bundles downloaded via the https source are `.tgz` files, and we should return the ungzip'ed untar'd representation, like this: https://github.com/operator-framework/rukpak/blob/3b832db489499105211cda18f2217aad89e2bb8f/internal/storage/http.go#L84-L88

That way, if someone stored a tgz plain or registry bundle on an HTTP server somewhere, this http source could be used just like the other sources (all of which actually return an already unpacked view of the bundle)

Reply:
I see.  I'll change it.

Reply:
done.